src/compiler.ml

```diff
@@ -1233,15 +1328,15 @@ let prechr_rule_of_ast depth macros state r =
   { pto_match; pto_remove; pguard; pnew_goal; pamap; pname; pifexpr; pcloc }
   
 (* used below *)
-let preterms_of_ast ?on_type loc ~depth macros state f t =
+let of_ast transformer loc ~depth macros state f t =
```

Comment:
of_ast aims to take a raw_term, type_abbrev or type_expr and clean it.
There should be specialized version for type_abbrev and type_expr, if we want to keep modes attached to types